### PR TITLE
CheckpointManager: migrate `checkpoints` map to `java.time` API

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/CheckpointManager.java [hunk 1]
```diff
@@ -144,11 +146,11 @@ private Sha256Hash readBinary(InputStream inputStream) throws IOException {
                     throw new IOException("Incomplete read whilst loading checkpoints.");
                 StoredBlock block = StoredBlock.deserializeCompact(params, buffer);
                 ((Buffer) buffer).position(0);
-                checkpoints.put(block.getHeader().getTimeSeconds(), block);
+                checkpoints.put(block.getHeader().getTimeInstant(), block);
             }
             Sha256Hash dataHash = Sha256Hash.wrap(digest.digest());
             log.info("Read {} checkpoints up to time {}, hash is {}", checkpoints.size(),
-                    TimeUtils.dateTimeFormat(checkpoints.lastEntry().getKey() * 1000), dataHash);
+                    TimeUtils.dateTimeFormat(checkpoints.lastEntry().getKey().toEpochMilli()), dataHash);
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/CheckpointManager.java:153]: We should (separately) add a `TimeUtils.dateTimeFormat(Instant)`

File: core/src/main/java/org/bitcoinj/core/CheckpointManager.java [hunk 2]
```diff
@@ -181,11 +183,11 @@ private Sha256Hash readTextual(InputStream inputStream) throws IOException {
                 buffer.put(bytes);
                 ((Buffer) buffer).position(0);
                 StoredBlock block = StoredBlock.deserializeCompact(params, buffer);
-                checkpoints.put(block.getHeader().getTimeSeconds(), block);
+                checkpoints.put(block.getHeader().getTimeInstant(), block);
             }
             HashCode hash = hasher.hash();
             log.info("Read {} checkpoints up to time {}, hash is {}", checkpoints.size(),
-                    TimeUtils.dateTimeFormat(checkpoints.lastEntry().getKey() * 1000), hash);
+                    TimeUtils.dateTimeFormat(checkpoints.lastEntry().getKey().toEpochMilli()), hash);
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/CheckpointManager.java:190]: 2nd potential usage of  `TimeUtils.dateTimeFormat(Instant)`

File: core/src/main/java/org/bitcoinj/core/CheckpointManager.java [hunk 3]
```diff
@@ -222,24 +230,30 @@ public Sha256Hash getDataHash() {
...
         checkArgument(!(store instanceof FullPrunedBlockStore), "You cannot use checkpointing with a full store.");
 
-        timeSecs -= 60 * 60 * 24 * 7; // one week in seconds
+        time = time.minus(7, ChronoUnit.WEEKS);
 
-        checkArgument(timeSecs > 0);
-        log.info("Attempting to initialize a new block store with a checkpoint for time {} ({})", timeSecs,
-                TimeUtils.dateTimeFormat(timeSecs * 1000));
+        log.info("Attempting to initialize a new block store with a checkpoint for time {} ({})", time.getEpochSecond(),
+                TimeUtils.dateTimeFormat(time.toEpochMilli()));
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/core/CheckpointManager.java:244]: 3rd potential usage of  `TimeUtils.dateTimeFormat(Instant)`

**Review by msgilligan — APPROVED** [review]
Looks good!

I left minor comments pointing out three places where a `TimeUtils.dateTimeFormat(Instant)` method would simplify the code. This would be a separate PR.

File: core/src/main/java/org/bitcoinj/core/CheckpointManager.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/core/CheckpointManager.java:153]: Yes, I have that commit on a branch already, but decided to PR it later.

**Comment by schildbach** [comment]
Merged.

**Pull request merged**